utf8proc.c

```diff
@@ -165,12 +168,9 @@ UTF8PROC_DLLEXPORT utf8proc_ssize_t utf8proc_encode_char(utf8proc_int32_t uc, ut
     dst[0] = 0xC0 + (uc >> 6);
     dst[1] = 0x80 + (uc & 0x3F);
     return 2;
-  } else if (uc == 0xFFFF) {
-    dst[0] = 0xFF;
-    return 1;
-  } else if (uc == 0xFFFE) {
-    dst[0] = 0xFE;
-    return 1;
+  } else if (((utf8proc_uint32_t)uc)-0xd800 < 0x800) {
+    // Surrogates not valid code point, can't be encoded!
+    return 0;
```

Comment:
OK, I see your point... (it doesn't matter to me as much anymore, because I now would like to just cut Julia's dependence on utf8proc.c anyway... make it pure Julia code if possible, with maybe just a C data file... [but not the one from utf8proc_data.c]).   I'll change the surrogate part back.